File: 프로그래머스/lv2/60057. 문자열 압축/문자열 압축.java
class Solution {
public static int solution(String s) {
        int answer = s.length();
        for (int i = 1; i <= s.length(); i++) {
            int num = i * 2;                       // 다음 범위 마지막 숫자
            int total = s.length();              // 원래 s 문자열 길이
            String word = s.substring(0, i);      // 첫번째 비교할 문자열
            int repeat = 0;                      // 반목한 숫자

            while (num <= s.length()) {
                if (word.equals(s.substring(num - i, num))) {
                    repeat += 1;
                } else {
                    word = s.substring(num - i, num);
                    if (repeat != 0) {
                        total -= (repeat * i - Integer.toString(repeat + 1).length());
                        repeat = 0;
                    }
                }
                num += i;
            }
            if (repeat != 0) {
                total -= (repeat * i - Integer.toString(repeat + 1).length());
                repeat = 0;
            }
            if (answer > total) {
                answer = total;
            }
        }
        return answer;
    }
}

File: 프로그래머스/lv2/60057. 문자열 압축/문자열 압축.py
def solution(s):
    answer = 1000000000000
    if len(s) == 1:
        answer = 1
    for i in range(1,len(s)):
        total = len(s)
        num = 2*i
        word = s[:i]
        repeat = 0
        while num <= len(s):
            # print('i',s[num-i:num])
            if word == s[num-i:num]:
                repeat += 1
            else:
                word = s[num-i:num]
                if repeat != 0:
                    if repeat > 8:
                        total -= (repeat*i-2)
                    else:
                        total -= (repeat*i-1)
                    repeat = 0
            num += i
        if repeat > 0:
            if repeat > 8:
                total -= (repeat*i-2)
            else:
                total -= (repeat*i-1)
        
        if  answer > total:
            answer = total
    return answer
